infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntimeTest.java

```diff
@@ -366,10 +427,34 @@ public void testRepublishContainerOutputAsMachineLogEvents() throws Exception {
     assertTrue(captor.getAllValues().containsAll(machineLogs));
   }
 
+  @Test
+  public void testHandleUnrecoverableEvent() throws Exception {
+    final UnrecoverableEventHanler unrecoverableEventHanler =
```

Comment:
Please, test unrecoverable events matching/non-matching in the whole flow instead of with a separated component. We don't have any guarantees that handler is called as it is supposed in this test case.
Please cover cases:
non-matching event
matching by the reason field
matching by the message field

It would be nice to verify that stop of the runtime is really called after appearing of the matching event.

Reply:
@garagatyi was not able to figure out how exactly to test unrecoverable events in starting flow, but have added separate tests for matching by the reason / matching by the message. Also, not sure how exactly verifying that stop of the runtime e.g. `internalStop` should be tested - currently done by verifying that namespace cleanup was called

Are you ok to merge it in the current state, since I do not really want to face merge conflicts, and we could discuss more tests cases which I would add later in separate PR ?